<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>HDTit</class>
 <widget class="QMainWindow" name="HDTit">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>709</width>
    <height>524</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>HDT-it !</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="verticalLayoutWidget">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Triple Search Pattern</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <layout class="QGridLayout" name="gridLayout_2" rowstretch="0,0,0,0,0">
             <item row="1" column="1">
              <widget class="QLineEdit" name="subjectPatternEdit"/>
             </item>
             <item row="2" column="1">
              <widget class="QLineEdit" name="predicatePatternEdit"/>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="subjectLabel">
               <property name="text">
                <string>S</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="predicateLabel">
               <property name="text">
                <string>P</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QLineEdit" name="objectPatternEdit"/>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="objectLabel">
               <property name="text">
                <string>O</string>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QPushButton" name="allSubjectButton">
               <property name="text">
                <string>All</string>
               </property>
              </widget>
             </item>
             <item row="2" column="2">
              <widget class="QPushButton" name="allPredicateButton">
               <property name="text">
                <string>All</string>
               </property>
              </widget>
             </item>
             <item row="3" column="2">
              <widget class="QPushButton" name="allObjectButton">
               <property name="text">
                <string>All</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QPushButton" name="searchPatternButton">
                 <property name="text">
                  <string>Search</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="randomTripleButton">
                 <property name="text">
                  <string>Random Triple</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QTabWidget" name="tabWidget">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab">
           <attribute name="title">
            <string>Predicates</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLabel" name="numPredicateCount">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="predicateCountSlider">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::NoTicks</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QPushButton" name="selectAllPredicatesButton">
                <property name="text">
                 <string>All</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="selectNoneButton">
                <property name="text">
                 <string>None</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QListView" name="listView"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_2">
           <attribute name="title">
            <string>Stats</string>
           </attribute>
           <layout class="QHBoxLayout" name="horizontalLayout_3"/>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="MatrixViewWidget" name="matrixView" native="true"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>709</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpenHDT"/>
    <addaction name="actionSaveHDT"/>
    <addaction name="separator"/>
    <addaction name="actionImportRDF"/>
    <addaction name="actionExportRDF"/>
   </widget>
   <widget class="QMenu" name="menuViews">
    <property name="title">
     <string>Views</string>
    </property>
    <addaction name="actionFrontView"/>
    <addaction name="actionLeftView"/>
    <addaction name="actionTopView"/>
    <addaction name="action3Dview"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuViews"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpenHDT"/>
   <addaction name="actionSaveHDT"/>
   <addaction name="separator"/>
   <addaction name="actionImportRDF"/>
   <addaction name="actionExportRDF"/>
   <addaction name="separator"/>
   <addaction name="actionViews"/>
   <addaction name="actionFrontView"/>
   <addaction name="actionLeftView"/>
   <addaction name="actionTopView"/>
   <addaction name="action3Dview"/>
   <addaction name="separator"/>
   <addaction name="actionTest"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpenHDT">
   <property name="text">
    <string>Open HDT</string>
   </property>
  </action>
  <action name="actionImportRDF">
   <property name="text">
    <string>Import RDF</string>
   </property>
  </action>
  <action name="actionSaveHDT">
   <property name="text">
    <string>Save HDT</string>
   </property>
  </action>
  <action name="actionExportRDF">
   <property name="text">
    <string>Export RDF</string>
   </property>
  </action>
  <action name="actionViews">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Views:</string>
   </property>
  </action>
  <action name="actionFrontView">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Front</string>
   </property>
   <property name="toolTip">
    <string>Front View</string>
   </property>
  </action>
  <action name="actionLeftView">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Left</string>
   </property>
   <property name="toolTip">
    <string>Left View</string>
   </property>
  </action>
  <action name="actionTopView">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Top</string>
   </property>
   <property name="toolTip">
    <string>Top View</string>
   </property>
  </action>
  <action name="action3Dview">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>3D</string>
   </property>
   <property name="toolTip">
    <string>3D View</string>
   </property>
  </action>
  <action name="actionTest">
   <property name="text">
    <string>Test</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MatrixViewWidget</class>
   <extends>QWidget</extends>
   <header>matrixviewwidget.hpp</header>
   <slots>
    <signal>numMaxPredicateCountChanged(int)</signal>
    <signal>predicateRangeChanged(int,int)</signal>
    <signal>maxPredicateCountChanged(int)</signal>
    <slot>updateGL()</slot>
    <slot>selectAllPredicates()</slot>
    <slot>selectNonePredicates()</slot>
    <slot>setMinimumPredicateCount(int)</slot>
    <slot>set3Dview()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>selectAllPredicatesButton</sender>
   <signal>clicked()</signal>
   <receiver>matrixView</receiver>
   <slot>selectAllPredicates()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>365</y>
    </hint>
    <hint type="destinationlabel">
     <x>462</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>selectNoneButton</sender>
   <signal>clicked()</signal>
   <receiver>matrixView</receiver>
   <slot>selectNonePredicates()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>378</x>
     <y>365</y>
    </hint>
    <hint type="destinationlabel">
     <x>462</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>allSubjectButton</sender>
   <signal>clicked()</signal>
   <receiver>subjectPatternEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>357</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>287</x>
     <y>105</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>allPredicateButton</sender>
   <signal>clicked()</signal>
   <receiver>predicatePatternEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>353</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>303</x>
     <y>141</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>allObjectButton</sender>
   <signal>clicked()</signal>
   <receiver>objectPatternEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>349</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>323</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>predicateCountSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>matrixView</receiver>
   <slot>setMinimumPredicateCount(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>142</x>
     <y>342</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>matrixView</sender>
   <signal>maxPredicateCountChanged(int)</signal>
   <receiver>predicateCountSlider</receiver>
   <slot>setMaximum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>457</x>
     <y>167</y>
    </hint>
    <hint type="destinationlabel">
     <x>332</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>predicateCountSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>numPredicateCount</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>330</y>
    </hint>
    <hint type="destinationlabel">
     <x>96</x>
     <y>303</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
